### PR TITLE
Add a base UnitActionType so save/load doesn't break static associations

### PR DESCRIPTION
**Pull request by ThislsAUsername**

### CODE REVIEW

File: src/Engine/UnitActionType.java [hunk 1]
```diff
@@ -25,9 +25,35 @@
...
+    @Override
+    public boolean equals(Object obj)
+    {
+      if( this == obj )
+        return true;
+      if( obj == null )
+        return false;
+      if( getClass() != obj.getClass() )
+        return false;
+      UnitActionType other = (UnitActionType) obj;
```
Comment (ThatIsAUsername) [inline src/Engine/UnitActionType.java]: Probably worth checking that `null != other` as well, in case the cast fails.

**Comment by ThatIsAUsername** [comment]
Pros: Simple to implement, works for current cases.
Cons: Potentially fragile. Will fail if anyone ever does a save/load and compares using `==`, and it wont' be obvious why.

File: src/Engine/UnitActionType.java (hunk 1 above)
Comment (ThislsAUsername) [reply src/Engine/UnitActionType.java]: Hmm.
We just checked that the classes are equal, and I'm casting to an interface that this class implements.
I can't see a case where that could happen.

Comment (ThatIsAUsername) [reply src/Engine/UnitActionType.java]: Ah. Right. I blame being awake at 5 am.

**Pull request closed without merging**